pkg/substitution/substitution.go

```diff
@@ -25,8 +25,12 @@ import (
 	"knative.dev/pkg/apis"
 )
 
-const parameterSubstitution = `.*?(\[\*\])?`
-const braceMatchingRegex = "(\\$(\\(%s(\\.(?P<var1>%s)|\\[\"(?P<var2>%s)\"\\]|\\['(?P<var3>%s)'\\])\\)))"
+const (
+	parameterSubstitution = `.*?(\[\*\])?`
+
+	// BraceMatchingRegex is a regex for parameter references including dot notation, bracket notation with single and double quotes.
+	BraceMatchingRegex = "(\\$(\\(%s(\\.(?P<var1>%s)|\\[\"(?P<var2>%s)\"\\]|\\['(?P<var3>%s)'\\])\\)))"
```

Comment:
could you avoid exporting this, and instead create an exported helper function?

Reply:
Done!

Reply:
I think the purpose of this package is to validate and parse parameters (it's hard to tell haha-- substitution is a bit messy), so the reason I suggest writing a helper function is to think about what validation and parsing can be abstracted away. The functions exported by a package form a sort of API that other packages can use. Imagine if you encountered an API with this getter function-- I think it would be confusing. This is where a package just for params would be really helpful! Since we don't have that, could you try to write a helper function that's consistent with what the `substitution` package is designed for? I am hoping to avoid the use of regexes in many places in the codebase because they are very hard to understand.

Reply:
Thank you @lbernick for the explanation. That makes sense to me! Done.